Applications/Spire/Source/Ui/Checkbox.cpp

```diff
@@ -56,99 +27,98 @@ Checkbox::Checkbox(QWidget* parent)
 Checkbox::Checkbox(const QString& label, QWidget* parent)
     : QCheckBox(label, parent),
       m_is_read_only(false) {
-  setAttribute(Qt::WA_Hover);
-  auto font = QFont("Roboto");
-  font.setPixelSize(scale_height(12));
-  setFont(font);
-}
-
-void Checkbox::keyPressEvent(QKeyEvent* event) {
-  if(m_is_read_only) {
-    event->accept();
-    return;
-  }
-  QCheckBox::keyPressEvent(event);
-}
-
-void Checkbox::mousePressEvent(QMouseEvent* event) {
-  event->accept();
-  if(m_is_read_only) {
-    return;
-  }
-  if(event->button() == Qt::LeftButton) {
-    setChecked(!isChecked());
-  }
-}
-
-void Checkbox::paintEvent(QPaintEvent* event) {
-  auto painter = QPainter(this);
-  painter.fillRect(QRect(get_box_position(), BOX_SIZE()), get_box_color());
-  painter.fillRect(get_inner_box_rect(), get_inner_box_color());
-  if(isChecked()) {
-    auto icon = QPixmap::fromImage(CHECK_ICON());
-    auto image_painter = QPainter(&icon);
-    image_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
-    image_painter.fillRect(icon.rect(), get_check_color());
-    painter.drawPixmap(get_check_position(), icon);
-  }
-  if(!text().isEmpty()) {
-    painter.setPen(TEXT_COLOR());
-    painter.drawText(QRect(get_text_position(), size()), 0, text());
-  }
+  setObjectName("Checkbox");
+  setStyleSheet(QString(R"(
+    #Checkbox {
+      background-color: #00000000;
+      color: #000000;
+      font-family: "Roboto";
+      font-size: %1px;
+      outline: none;
+    }
+    #Checkbox::indicator {
+      background-color: #FFFFFF;
+      border: %2px solid #C8C8C8;
+      height: %3px;
+      width: %4px;
+    }
+    #Checkbox::indicator:hover {
+      border-color: #4B23A0;
+    }
+    #Checkbox::indicator:focus {
+      border-color: #4B23A0;
+    }
+    #Checkbox::indicator:read-only {
+      background-color: #00000000;
+      border-color: #00000000;
+    }
+    #Checkbox:disabled {
+      color: #C8C8C8;
+    }
+    #Checkbox::indicator:disabled {
+      background-color: #F5F5F5;
+    }
+    #Checkbox::indicator:read-only:disabled {
+      background-color: #00000000;
+      border-color: #00000000;
+    })").arg(scale_width(12)).arg(scale_width(1)).
```

Comment:
I think this case is an exception to the operator rule. You can put the period on line 64 before the call to arg(...) on line 65.

Reply:
I need to update the style guide, but the ```.``` should go at the end of the line instead of at the beginning.

Reply:
Okay, we'll have to update this as we come across it because it's pretty prevalent.

Reply:
Yep, very true.